### ISSUE TITLE
Using `title` as an alias for the `CreativeWork.headline` property

### ISSUE THREAD

**Issue by nokome** [stencila/schema#219]
We currently use `title` as an alias for the `headline` of a `CreativeWork`:

https://github.com/stencila/schema/blob/5d6c75ab1e39263e1feb4bf024baf72fb31dea2a/schema/CreativeWork.schema.yaml#L170-L187

This manifests in the [JSON-LD context](https://schema.stenci.la/stencila.jsonld) as this mapping:

```jsonld
    "title": {
      "@id": "schema:headline"
    },
```

@mbjones pointed out that other schemas have mapped `title` to `schema:name` which may cause ambiguities. I looked into this a little.

I think we chose to use `title` as an alias of schema.org `headline`, rather than `name`, mainly because Google's structured data guidelines [recommend (actually, _require_) `headline`](https://developers.google.com/search/docs/data-types/article#article-types). So, `headline` seemed to be the best property to translate, say a JATS [`<article-title>`](https://jats.nlm.nih.gov/archiving/tag-library/1.1d1/n-e630.html) element, to for an `Article` entity.

The [CodeMeta JSON-LD](https://github.com/codemeta/codemeta/blob/master/codemeta.jsonld) does not show any mapping from schema.org `name` to a `title` property. However, at least one of the CodeMeta crosswalks maps a "title" property from a software package to the `name` property of a `SoftwareSourceCode` entity. e.g. https://github.com/codemeta/codemeta/blob/master/crosswalks/R%20Package%20Description.csv#L46 .

I don’t think there is a conflict here with what we are doing for `Article`. For `SoftwareSourceCode`, `name` seems like the most appropriate property, for `Article`, `headline` seems best.

Thoughts appreciated.

**Comment by mbjones** [comment]
Thanks, @nokome . We did end up using `schema:name` in our json-ld context for CodeMeta as our property intended to hold a title (`"name": { "@id": "schema:name"},`). From the CodeMeta perspective, we were trying to map many software metadata dialects onto schema.org. At first, we considered a custom `title` field because we didn't think it was covered in schema.org. After feedback, many people indicated that `so:name` is the proper holder of a title-like string, and it can be applied to any CreativeWork (originally, I think `headline` was only applicable to `schema:NewsArticle`). So we settled on `schema:name` and we mapped the title field from many dialects (e.g., ([`datacite:title`](https://github.com/codemeta/codemeta/blob/master/crosswalks/DataCite.csv#L46), [`dcterms:title`](https://github.com/codemeta/codemeta/blob/master/crosswalks/DCAT.csv#L46), [`R:Title`](https://github.com/codemeta/codemeta/blob/master/crosswalks/R%20Package%20Description.csv#L46),  etc) to `schema:name`. Now, applications we build expect to be able to use `schema:name` as the equivalent of a title to be displayed for a piece of software or for data.  

Related discussions as us to use of `so:name` versus `so:headline`, most of which conclude that name is preferred over headline:

- https://github.com/schemaorg/schemaorg/pull/423 (and in particular, [this comment](https://github.com/schemaorg/schemaorg/pull/423#issuecomment-93999467))
- https://github.com/schemaorg/schemaorg/issues/205
- https://github.com/ampproject/amphtml/issues/483
- https://stackoverflow.com/questions/38000771/what-is-the-name-property-in-schema-org

I'm not recommending anything in particular for the Stencila schema, but I was wondering how your apps dealt with the redundancy. Do you search for both?

**Comment by nokome** [comment]
Thanks @mbjones for taking the time to write that up and provide those links. Very useful and much appreciated.

To me they suggest that we should use `so:name` instead of `so:headline`, particularly given that several people have suggested deprecating `headline`.

We do check for both `name` and `title` in dialects. Most of the time `name` gets mapped to `so:name`, and `title` gets mapped to  `stencila:title` (which is an alias for `so:headine`) if the entity is an `Article` (e.g. https://github.com/stencila/encoda/blob/a27cb25371444e95f0100c8a4f059e601e179773/src/codecs/jats/index.ts#L452) and to `so:name` if it is `SoftwareSourceCode` (e.g. https://github.com/stencila/dockta/blob/e8949317d91ca77f67beed0608b761f9515b7c42/src/DockerParser.ts#L41).

If we aliased `stencila:title` to `so:name` instead then there would be greater consistency and it shouldn't require any code changes, just a one line change in `CreativeWork.schema.yaml` (I think).

**Comment by thadguidry** [comment]
@nokome Headline and alternateHeadline are used in the News domain for the headlining title of an article or piece of writing.  It won't be deprecated on Schema.org that's for sure, but for Stencila it's probably not as useful for research articles or papers, as much as the properties of name, and alternateName would be.

**Comment by nokome** [comment]
Yep, agreed. I liked this quote from https://github.com/schemaorg/schemaorg/pull/423#issuecomment-93999467. I think it petty much hits the nail on the head with what we did here:

> Without looking, I can pretty much guarantee that there will be some usage
of headline on non-NewsArticle classes in the wild. But that's because
people look for "title", don't find it, and then grasp at the next closest
thing they find defined on CreativeWork (headline), rather than falling
back to Thing and name.

**Comment by thadguidry** [comment]
@nokome What people sometimes incorrectly do, and what you should really do, will sometimes differ. :-)  I can help to ensure you are making the right choices, feel free to ask anytime.  You will find plenty of incorrect usage of Schema.org in the wild.
